R/vpc_tte.R

```diff
@@ -186,152 +134,55 @@ vpc_tte <- function(sim = NULL,
-        msg("Warning: vpc_tte() expected the observed dependent variable to contain only 0 (censored, or no event observed) or 1 (event observed). Setting all observations != 1 to 0.", verbose)
-      } else {
-        obs[obs$dv != 1,]$dv <- 1 # some people use DV to indicate the event time.
-        msg("Warning: vpc_tte() expected the dependent variable to contain only 0 (censored, or no event observed) or 1 (event observed). Setting all observations != 1 to 1.", verbose)
-      }
-    }
-    if("cens" %in% tolower(colnames(obs))) { # some people use a 'cens' column to indicate censoring
-      msg(paste0("Detected column '",colnames(obs)[match("cens", tolower(colnames(obs)))],"' with censoring information in observation data, assuming 1=censored event, 0=observed event. Please transpose data if assumption not correct."), TRUE)
-      colnames(obs)[match("cens", tolower(colnames(obs)))] <- "cens"
-      obs[obs$cens == 1,]$dv <- 0
-    }
-    if(rtte) {
-      if(rtte_calc_diff) {
-        obs <- relative_times(obs)
-      }
-      obs <- obs %>%
-        dplyr::group_by_("id") %>%
-        dplyr::arrange_("id", "t") %>%
-        dplyr::mutate(rtte = 1:length(dv))
-#       obs %>% dplyr::group_by(id) %>% dplyr::mutate(rtte = cumsum(dv != 0))
-#       obs[obs$dv == 0,]$rtte <- obs[obs$dv == 0,]$rtte + 1 # these censored points actually "belong" to the next rtte strata
-      stratify_pars <- c(stratify_pars, "rtte")
-    } else {
-      obs <- obs %>%
-        dplyr::group_by_("id") %>%
-        dplyr::mutate(last_obs = 1*(1:length(time) == length(time)), repeat_obs = 1*(cumsum(dv) > 1)) %>%
-        dplyr::filter(dv == 1 | last_obs == 1) %>%
-        dplyr::filter(!duplicated(id))
-      obs$rtte <- 1
-    }
-    
-    # add stratification column and comput KM curve for observations
-    obs <- add_stratification(obs, stratify_pars)
+    obs_data <-
+      format_vpc_input_data_tte(
+        dat=obs,
+        cols=vpc_data$cols$obs,
+        stratify=stratify_pars,
+        rtte=rtte,
+        rtte_calc_diff=rtte_calc_diff,
+        what="observed",
+        verbose=verbose
+      )
+    obs <- obs_data$dat
+    stratify_pars <- obs_data$stratify
+
+    # add stratification column and compute KM curve for observations
     if(!is.null(kmmc) && kmmc %in% names(obs)) {
       obs_km <- compute_kmmc(obs, strat = "strat", reverse_prob = reverse_prob, kmmc=kmmc)
     } else {
-      if(show$obs_ci) {
-        if(length(ci) == 2 && (round(ci[1],3) != round((1-ci[2]),3))) {
-          stop("Sorry, only symmetric confidence intervals can be computed. Please adjust the ci argument.")
-        }
-        obs_km <- compute_kaplan(obs, strat = "strat", reverse_prob = reverse_prob, rtte_conditional = rtte_conditional, ci = ci)
-      } else {
-        obs_km <- compute_kaplan(obs, strat = "strat", reverse_prob = reverse_prob, rtte_conditional = rtte_conditional)
-      }
+      # TODO: Review 2021-04: Always computing obs_km when obs is available
```

Comment:
It seems like the only reason to still do compute_kaplan() on obs data when `obs_ci=FALSE` is to get the bins for the VPC. Perhaps add as a comment?

Reply:
I can imagine using it for getting the bins, but I'm not sure what you're asking for as the comment.  Are you simply wanting a comment where the "TODO" is saying "It seems like the only reason to still do compute_kaplan() on obs data when obs_ci=FALSE is to get the bins for the VPC"?  (That seems straight-forward, but I'd rather be sure that I'm interpreting it correctly rather than lose the intent by making a change that may not be the requested change.)

Reply:
Yeah, just a clarification to ourselves about *why* we're still using obs data, the current comment only says describes the *what*. E.g. just "Obs data is still used to calculate bins, even though obs data not requested in plot." or something like that.